packages/admin-panel/src/VizBuilderApp/api/mutations/useSaveVisualisation.js

```diff
@@ -4,20 +4,22 @@
  *
  */
 import { useMutation } from 'react-query';
-import { post } from '../api';
+import { post, put } from '../api';
 
-// Todo: Finish Save Button @see https://linear.app/tupaia/issue/MEL-20/save-button
 export const useSaveVisualisation = config =>
   useMutation(
-    ['saveDashboardVisualisation', config],
-    () =>
-      post('saveDashboardVisualisation', {
+    ['dashboardVisualisation', config],
+    () => {
+      if (config.id) {
+        return put(`dashboardVisualisation/${config.id}`, {
```

Comment:
I've changed the save route (save an existing dashboard viz) to /PUT dashboardVisualisation/:dashboardVisualisationId to make it a bit more resty